dist/common/scripts/scylla_util.py

```diff
@@ -121,15 +122,70 @@ def curl(url, headers=None, byte=False, timeout=3, max_retries=5, retry_interval
                 raise
 
 
-class gcp_instance:
+class cloud_instance(metaclass=ABCMeta):
+    @abstractmethod
+    def get_local_disks(self):
+        pass
+
+    @abstractmethod
+    def get_remote_disks(self):
+        pass
+
+    @abstractmethod
+    def private_ipv4(self):
+        pass
+
+    @abstractmethod
+    def is_supported_instance_class(self):
+        pass
+
+    @abstractmethod
+    def instance_class(self):
+        pass
+
+    @abstractmethod
+    def instance_size(self):
+        pass
+
+    @abstractmethod
+    def io_setup(self):
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def check():
+        pass
+
+    @property
+    @abstractmethod
+    def user_data(self):
+        pass
+
+    @property
+    @abstractmethod
+    def nvmeDiskCount(self):
```

Comment:
Should this be local_disk_count()? Do we want to count nvme vs non-nvme, or local vs. remote?

Reply:
Ah, you are correct.
This method is used for scylla_io_setup to count ephemeral disks, but this count include both ephemeral and persistent, the code is wrong.
Instead, we should drop current method and provide local_disk_count() and remote_disk_count().